src/Aspire.Hosting.Kubernetes/Generators/HelmOutputGenerator.cs

```diff
@@ -0,0 +1,392 @@
+        }
+
+        // This service's sub-dict
+        servicesDict[svcName] = new Dictionary<string, object>
+        {
+            [KubernetesPublisherManifestKeys.Type] = svcType,
+            [KubernetesPublisherManifestKeys.Port] = svcPort,
+        };
+    }
+
+    private static Dictionary<string, object> AssignConfigMapData(V1ConfigMap configMap)
+    {
+        var dataSection = new Dictionary<string, object>();
+
+        if (configMap.Data != null)
+        {
+            foreach (var kvp in configMap.Data)
+            {
+                dataSection[kvp.Key] = kvp.Value;
+            }
+        }
+
+        return dataSection;
+    }
+
+    private static Dictionary<string, object> AssignSecretStringData(V1Secret secret, Dictionary<string, object> valuesObj, string secName)
+    {
+        var stringDataDict = new Dictionary<string, object>();
+
+        // Ensure .Values.secrets exists
+        if (!valuesObj.TryGetValue(KubernetesPublisherManifestKeys.Secrets, out var secretsObj))
+        {
+            secretsObj = new Dictionary<string, object>();
+            valuesObj[KubernetesPublisherManifestKeys.Secrets] = secretsObj;
+        }
+        var secretsDict = (Dictionary<string, object>)secretsObj;
+
+        // Create or overwrite this secret's sub-dict in .Values.secrets
+        var secretSubDict = new Dictionary<string, object>();
+        secretsDict[secName] = secretSubDict;
+
+        if (secret.StringData is { Count: > 0 })
+        {
+            // For each secret key, store the default in .Values.secrets.<secName>.<key>
+            // and reference a Helm placeholder in stringData.
+            foreach (var (key, value) in secret.StringData)
+            {
+                var defaultValue = value ?? string.Empty;
+
+                secretSubDict[key] = defaultValue;
+                stringDataDict[key] = SecretValuePlaceholder(secName, key);
+            }
+        }
+
+        return stringDataDict;
+    }
+
+    private static void AssignContainerToImages(Dictionary<string, object> valuesObj, V1Container container, string containerName)
+    {
+        if (!valuesObj.TryGetValue(KubernetesPublisherManifestKeys.Images, out var imagesObj))
+        {
+            imagesObj = new Dictionary<string, object>();
+            valuesObj[KubernetesPublisherManifestKeys.Images] = imagesObj;
+        }
+
+        if (imagesObj is not Dictionary<string, object> imagesDict)
+        {
+            throw new DistributedApplicationException("Expected .Values.images to be a dictionary.");
+        }
+
+        var imageRepo = container.Image ?? throw new DistributedApplicationException("Container image is null, and cannot be published.");
+        var imageTag = "latest";
+        if (imageRepo.Contains(':'))
+        {
+            var idx = imageRepo.LastIndexOf(':');
+            imageTag = imageRepo[(idx + 1)..];
+            imageRepo = imageRepo[..idx];
+        }
```

Comment:
Is this usable:

https://github.com/dotnet/aspire/blob/7834ae88d2b3ed8895e9bbb058d21ffd3491ff56/src/Aspire.Hosting/Utils/ContainerReferenceParser.cs

Reply:
yep - cheers

Reply:
its internal unfortunately

Reply:
Shared source 😄

Reply:
Haha - I'll drop it in to each when i get to the image processing stage 👅

Reply:
https://github.com/dotnet/aspire/tree/main/src/Shared